packages/@aws-cdk/aws-redshift/test/table.test.ts

```diff
@@ -0,0 +1,135 @@
+import { Template } from '@aws-cdk/assertions';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as cdk from '@aws-cdk/core';
+import * as redshift from '../lib';
+
+describe('cluster table', () => {
+  const tableName = 'tableName';
+  const tableColumns = [{ name: 'col1', dataType: 'varchar(4)' }, { name: 'col2', dataType: 'float' }];
+
+  let stack: cdk.Stack;
+  let vpc: ec2.Vpc;
+  let cluster: redshift.ICluster;
+  let databaseProps: redshift.DatabaseProps;
+
+  beforeEach(() => {
+    stack = new cdk.Stack();
+    vpc = new ec2.Vpc(stack, 'VPC');
+    cluster = new redshift.Cluster(stack, 'Cluster', {
+      vpc: vpc,
+      vpcSubnets: {
+        subnetType: ec2.SubnetType.PUBLIC,
+      },
+      masterUser: {
+        masterUsername: 'admin',
+      },
+      publiclyAccessible: true,
+    });
+    databaseProps = {
+      cluster: cluster,
+      databaseName: 'databaseName',
+    };
+  });
+
+  it('creates using custom resource', () => {
+    new redshift.Table(stack, 'Table', {
+      ...databaseProps,
+      tableColumns,
+    });
+
+    Template.fromStack(stack).hasResourceProperties('Custom::RedshiftDatabaseQuery', {
+      tableName: 'Table',
+      tableColumns,
+    });
+  });
+
```

Comment:
Please add a test case that the default retention policy is "retain".

Reply:
Already exists below: https://github.com/aws/aws-cdk/pull/15931/files/10bb1e74594a8a101c863ce4a51684d8dbea805e#diff-c7482582e791ef6fe716be6652f15b2bd24a1ecdad97ace5d334215b4b37f96cR106